py/derivation.nix

```diff
@@ -0,0 +1,33 @@
+{
+  lib,
+  python311,
+  gitignoreSource,
+  mypycSupport ? false
+}:
+
+let
+  python = python311;
+  pythonPackages = python.pkgs;
+  runtimeDeps = with pythonPackages; [ setuptools pysdl2 ];
+  devDeps = with pythonPackages; [ mypy black ];
+in
+
+pythonPackages.buildPythonApplication rec {
+  name = "rosettaboy-py";
+  src = gitignoreSource ./.;
+
+  nativeBuildInputs = lib.optional mypycSupport pythonPackages.mypy;
+
+  passthru.python = python.withPackages (p: runtimeDeps ++ devDeps);
+  passthru.devTools = [ python ];
```

Comment:
Just a heads up: earlier `passthru.devtools` used `python.withPackages ...` while this just uses the unmodified python interpreter.

https://github.com/shish/rosettaboy/blob/2f7b52686962b453a438e956cbfc4429374c80f1/py/derivation.nix#L15-L25

---

I don't think this actually makes a difference since `devTools` is only used for the devShell and `propagatedBuildInputs` gets [picked up anyways by `mkShell`](https://github.com/NixOS/nixpkgs/blob/f3b7d6414b06eb493ed7f258a6938be440b7cdaf/pkgs/build-support/mkshell/default.nix#L38) (and even if the "bare" interpreter gets put on `$PATH` ahead of the one including the packages it's still fine since the interpreter's [setup hook](https://github.com/NixOS/nixpkgs/blob/6979a72840a1f6f034cb6a11a3d114b17f792ad4/pkgs/development/interpreters/python/setup-hook.sh#L15-L17) will make sure they're added to `$PYTHONPATH`; it's only in cases where `$PYTHONPATH` is cleared or not passed to child processes that invoke `python` that there's an observable difference afaik).

---

I think it's probably fine to drop `devTools` from this derivation.

Reply:
Fails in CI after I removed the override in flake's devShells: https://github.com/shish/rosettaboy/actions/runs/4140614710/jobs/7159436299 :shrug: Just going to leave it for now.